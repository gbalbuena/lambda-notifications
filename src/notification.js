import notificationService from './services/notificationService';
import templateService from './services/templateService';

import { successResponse, runWarm } from './utils';

const notification = (event, context, callback) => {
  const { template, fields } = JSON.parse(event.body);
  const content = templateService.getContent(template, fields);

  const response = successResponse({
    status: notificationService.send(content),
    content,
  });

  callback(null, response);
};

export default runWarm(notification);
