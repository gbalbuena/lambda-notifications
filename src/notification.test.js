import notification from './notification';

describe('notification', () => {
  it('executes as expected', () => {
    const cb = jest.fn();

    const event = {
      body: '{"template": "Hi {name}", "fields": { "name": "John" } }',
    };
    notification(event, {}, cb);
    expect(cb).toBeCalled();
    expect(cb).toMatchSnapshot();
  });
});
