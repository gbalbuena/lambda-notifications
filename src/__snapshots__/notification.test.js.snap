// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`notification executes as expected 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      null,
      Object {
        "body": "{\\"status\\":\\"success\\",\\"content\\":\\"Hi John\\"}",
        "statusCode": 200,
      },
    ],
  ],
}
`;
